crates/sui-node/src/lib.rs

```diff
@@ -222,11 +214,8 @@ impl SuiNode {
                     // TODO: enable checkpoint sync on fullnode
                     // let metrics = CheckpointMetrics::new(&prometheus_registry);
                     // active_authority.sync_to_latest_checkpoint(&metrics).await?;
-                    (
-                        Some(active_authority.spawn_node_sync_process().await),
-                        None,
-                        None,
-                    )
+                    active_authority.respawn_node_sync_process().await;
+                    (None, None, None)
```

Comment:
I don't think I understand what you want me to change here - fullnode always starts sync process in the current code.

Reply:
The logic does not change, I'm suggesting we simplify the winding code

Reply:
sure - i tried to, and it doesn't seem to actually get much simpler. I can do this: https://github.com/MystenLabs/sui/commit/5f931e75eac1ecdde98561c6a7f04828ff8a0428 - but its not obvious if it can be made a ton simpler than that without major refactoring.

Reply:
I (somewhat) cleaned this up a tiny bit in https://github.com/MystenLabs/sui/pull/4233